### ISSUE TITLE
Check for max identifier length

### ISSUE THREAD

**Issue by subnix** [sbdchd/squawk#1098]
According to the [documentation](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS):
> The system uses no more than NAMEDATALEN-1 bytes of an identifier; longer names can be written in commands, but they will be truncated. By default, NAMEDATALEN is 64 so the maximum identifier length is 63 bytes.

Thus, If we use identifiers longer than 63 characters, they will be truncated. It would be great if squawk warned us about this in advance.

```sql
create table table_very_long_very_long_very_long_very_long_very_long_very_long (column_very_long_very_long_very_long_very_long_very_long_very_long bigint);
# NOTICE:  identifier "table_very_long_very_long_very_long_very_long_very_long_very_long" will be truncated to "table_very_long_very_long_very_long_very_long_very_long_very_lo"
# NOTICE:  identifier "column_very_long_very_long_very_long_very_long_very_long_very_long" will be truncated to "column_very_long_very_long_very_long_very_long_very_long_very_l"
# CREATE TABLE
```

**Comment by sbdchd** [comment]
Makes sense!

Really surprising that postgres just truncates the name 🫠

**Comment by sbdchd** [comment]
I wonder if this should be a lint rule or a syntax error from the parser. I suppose someone could tweak the `NAMEDATALEN` for their build and then disabling the rule would be handy.

**Comment by subnix** [comment]
> I wonder if this should be a lint rule or a syntax error from the parser

IMO this should be a lint rule, since PG accepts such identifiers.

**Comment by sbdchd** [comment]
This is now fixed in the latest release, 2.50.0:


```
warning[identifier-too-long]: `column_very_long_very_long_very_long_very_long_very_long_very_long` is too long and will be truncated to 63 bytes.
  ╭▸ stdin:1:81
  │
1 │ …ng_very_long_very_long_very_long (column_very_long_very_long_very_long_very_long_very_long_very_long bigint);
  │                                    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  ╭╴
1 - create table table_very_long_very_long_very_long_very_long_very_long_very_long (column_very_long_very_long_very_long_very_long_very_long_very_long bigint);
1 + create table table_very_long_very_long_very_long_very_long_very_long_very_long (column_very_long_very_long_very_long_very_long_very_long_very_l bigint);
  ╰╴

Find detailed examples and solutions for each rule at https://squawkhq.com/docs/rules
Found 3 issues in 1 file (checked 1 source file)
```

Thank you for filing the issue!
